### PR TITLE
Expose DNS conditions on the ARO Operator 

### PR DESCRIPTION
**Pull request by troy0820**
### Which issue this PR addresses:

Fixes [8858773](https://msazure.visualstudio.com/AzureRedHatOpenShift/_workitems/edit/8858773/)

### What this PR does / why we need it:

Places conditions of the DNS servers (custom DNS) on the ARO operator
### Test plan for issue:

E2E test will be performed to check all conditions on cluster create.  This is being done on PR https://github.com/Azure/ARO-RP/pull/1273
### Is there any documentation that needs to be updated for this PR?

No documentation.  The conditions are appended to the conditions that exist with the operator which include `MachineValid, InternetReachableFromMaster, InternetReachableFromWorker` and now `DNSValid`

### CODE REVIEW

File: pkg/operator/controllers/checker/auth.go [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+	}
+	creds.clientID = mysec.Data["azure_client_id"]
+	if _, ok := mysec.Data["azure_client_secret"]; !ok {
+		return nil, errors.New("azure_client_secret does not exists")
+	}
+	creds.clientSecret = mysec.Data["azure_client_id"]
+	if _, ok := mysec.Data["azure_tenant_id"]; !ok {
+		return nil, errors.New("azure_tenant_id does not exists")
+	}
+	creds.tenantID = mysec.Data["azure_client_id"]
```
Comment (troy0820) [inline pkg/operator/controllers/checker/auth.go:61]: Before returning, this should be base64 decoded? 🤔

Comment (bennerv) [reply pkg/operator/controllers/checker/auth.go:61]: Don't need to base64 decode it.  `.Data["value"]` will already be base64 decoded.

https://github.com/kubernetes/client-go/issues/198

**Comment by jim-minter** [comment]
@troy0820 #1296 has merged so we should not need to expose DNS conditions on the operator any more.  Are there other things in this PR that you do want to get merged?  If so, please update the PR; if not, please close.

**Comment by troy0820** [comment]
@jim-minter We can close this. If there are any other conditions that are necessary for the operator to add, I can make another branch.

**Pull request closed without merging**